languages/python/django-oso/tests/test_post_relationship.py

```diff
@@ -581,11 +823,46 @@ def test_reverse_many_relationship(tag_nested_many_many_fixtures):
         + ' FROM "test_app2_post"'
         + ' INNER JOIN "test_app2_user_posts"'
         + ' ON ("test_app2_post"."id" = "test_app2_user_posts"."post_id")'
-        + ' WHERE "test_app2_user_posts"."user_id" = 1'
+        + ' WHERE "test_app2_user_posts"."user_id" = 2'
     )
-    assert len(posts) == 4
 
 
+@pytest.mark.django_db
+def test_in_other_queryset_constraints_3(tag_nested_many_many_fixtures):
+    """Add constraints on partial."""
+    Oso.load_str(
+        """
+        allow(user: test_app2::User, "read", post: test_app2::Post) if
+            post.created_by in user.direct_reports and
+            # This constraint is invalid (cannot constrain the list directly)
+            # It never gets output because you get a *new* partial every time you
+            # access user.direct_reports
+            user.direct_reports.username = "foo" and
+            user.direct_reports.foo();
```

Comment:
@dhatch reading the comment on 838, it seems like this test is most concerned about the fact that we produce a *new* partial every time you access `user.direct_reports`, but I think that nuance is lost amidst the other issues with this example (namely, the `collection.username = "foo"` check is nonsensical and we don't support method calls on partials as the last line of the rule attempts to do).